diff --git a/tsr/models/isosurface.py b/tsr/models/isosurface.py
index bf62b8a..069db79 100644
--- a/tsr/models/isosurface.py
+++ b/tsr/models/isosurface.py
@@ -1,9 +1,9 @@
 from typing import Callable, Optional, Tuple
 
+import mcubes
 import numpy as np
 import torch
 import torch.nn as nn
-from torchmcubes import marching_cubes
 
 
 class IsosurfaceHelper(nn.Module):
@@ -18,7 +18,7 @@ class MarchingCubeHelper(IsosurfaceHelper):
     def __init__(self, resolution: int) -> None:
         super().__init__()
         self.resolution = resolution
-        self.mc_func: Callable = marching_cubes
+        self.mc_func: Callable = mcubes.marching_cubes
         self._grid_vertices: Optional[torch.FloatTensor] = None
 
     @property
@@ -41,12 +41,9 @@ class MarchingCubeHelper(IsosurfaceHelper):
         self,
         level: torch.FloatTensor,
     ) -> Tuple[torch.FloatTensor, torch.LongTensor]:
         level = -level.view(self.resolution, self.resolution, self.resolution)
-        try:
-            v_pos, t_pos_idx = self.mc_func(level.detach(), 0.0)
-        except AttributeError:
-            print("torchmcubes was not compiled with CUDA support, use CPU version instead.")
-            v_pos, t_pos_idx = self.mc_func(level.detach().cpu(), 0.0)
-        v_pos = v_pos[..., [2, 1, 0]]
+        vertices, triangles = self.mc_func(level.detach().cpu().numpy(), 0.0)
+        v_pos = torch.from_numpy(np.ascontiguousarray(vertices, dtype=np.float32))
+        t_pos_idx = torch.from_numpy(np.ascontiguousarray(triangles, dtype=np.int64))
         v_pos = v_pos / (self.resolution - 1.0)
         return v_pos.to(level.device), t_pos_idx.to(level.device)
